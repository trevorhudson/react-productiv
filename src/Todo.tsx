import React from "react";


interface TodoInterface {
  id: string,
  title: string,
  description: string,
  priority: string
}

interface PropsInterface  {
  todo: TodoInterface | null
}

/** Simple presentation component for a todo.
 *
 * Props:
 * - todo: like { id, title, description, priority }
 *
 * { EditableTodo, TopTodo } -> Todo
 **/


const Todo: React.FC<PropsInterface> = ({todo}) =>{
  return (
    <div className="Todo">
      <div><b>{todo!.title}</b> <small>{todo!.priority}</small></div>
      <div><small>{todo!.description}</small></div>
    </div>
  );
}

export default Todo;
