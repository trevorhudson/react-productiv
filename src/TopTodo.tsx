import React from "react";

import Todo from "./Todo";

/** Shows the top todo.
 *
 * Props:
 * - todos: array of [ {id, title, description, priority}, ...]
 *
 * TodoApp -> TopTodo
 */

function TopTodo({ todos }) {

  // lowest-priority # is the highest priority
  let top = todos.reduce(
    (acc, cur) => Number(cur.priority) < acc.priority ? cur : acc, todos[0]);

  return <Todo todo={top} />;
}

export default TopTodo;