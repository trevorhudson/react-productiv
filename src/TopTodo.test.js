
import { render } from "@testing-library/react";
import Todo from './Todo.js';
import TopTodo from './TopTodo.js';

describe('TopTodo Component tests', function () {
  const todos = [
    {
      id: 1,
      title: 'todo1',
      description: 'desc1',
      priority: 3
    },
    {
      id: 2,
      title: 'todo2',
      description: 'desc2',
      priority: 1
    },
    {
      id: 3,
      title: 'todo3',
      description: 'desc3',
      priority: 1
    }
  ];

  it('renders without crashing', function () {
    render(<TopTodo todos={todos} />);
  });

  it('passes snapshot test', function () {
    const { container } = render(<TopTodo todos={todos} />);
    expect(container).toMatchSnapshot();
  });

  it('expects single todo', function () {
    const { container } = render(<TopTodo todos={todos} />);
    expect(container.querySelectorAll('.Todo').length).toBe(1);
  });

  it('expects highest priority todo', function () {
    const { container } = render(<TopTodo todos={todos} />);
    expect(container.querySelector('.Todo')).toContainHTML('todo2');
  });
});
