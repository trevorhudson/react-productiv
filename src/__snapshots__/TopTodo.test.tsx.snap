// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopTodo Component tests passes snapshot test 1`] = `
<div>
  <div
    class="Todo"
  >
    <div>
      <b>
        todo2
      </b>
       
      <small>
        1
      </small>
    </div>
    <div>
      <small>
        desc2
      </small>
    </div>
  </div>
</div>
`;
